src/nccl_ofi_net.c

```diff
@@ -2518,6 +2525,39 @@ ncclResult_t nccl_net_ofi_isend(void *sendComm, void* data, int size,
 		goto error;
 	}
 
+	nccl_ofi_comp = sComm->baseComm.ofi_comp;
+	if (OFI_UNLIKELY(nccl_ofi_comp == NULL)) {
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("NCCL OFI component for dev %d is not initialised",
+			     sComm->dev);
+		goto error;
+	}
+
+	/*
+	 * In case, we are connecting to self, ensure that the request has
+	 * completed. If its completed, free the request. If not, progress the
+	 * function to process completions and return NULL request for send to
+	 * retry.
+	 */
+	if (OFI_UNLIKELY(sComm->connection_info && sComm->connection_info->connect_to_self == 1)) {
```

Comment:
What if we never call send with the this s_comm? Then should we wait for connection establishment in send_close()?

Reply:
At this time, we do not wait for any request to finish when closeSend() is called. I think we should do that but I will prefer to do it in a separate PR.